### PR TITLE
Extracted game logic into GamePlay file, makes gameLoop more readable

### LINKED ISSUE #32: Refactoring Index.js and add tests

### PR DESCRIPTION
**Pull request by erinnachen**
Seperated methods for drawing floors, bubbles, fruits, windup
Seperated methods for checking collisions windup-bubbles, dino-bubbles, dino-fruits, dino-windup
Loop now exits when dinosaur runs out of lives
Created methods to generate the collider and receiver objects for collision detection

closes #32

### CODE REVIEW

**Comment by erinnachen** [comment]
@adriennedomingus I've moved the logic out of index.js and now a bunch of methods live in game_play.js. I didn't get a chance to write new tests. Some areas for improvement:

* the game play file looks like it can be split into two files: one that has methods in charge of drawing and one that has methods involving collision detection. 
* right now the interfaces in the GamePlay methods are just oh so slightly different, I think they can be made more uniform/generic to DRY out the code.

File: lib/bubbles.js [hunk 1]
```diff
@@ -40,4 +40,11 @@ Bubble.prototype.move = function() {
   return this;
 };
 
+Bubble.prototype.fillUp = function() {
+  this.status = "floating";
+  this.filled = true;
+  this.height = 15;
+  this.width = 15;
+}
```
Comment (adriennedomingus) [inline lib/bubbles.js]: Cool. That'll help with collision

**Comment by adriennedomingus** [comment]
Index looks great. Yes, we can probably break game play up even further.

**Comment by adriennedomingus** [comment]
💪

**Pull request merged**